### PR TITLE
test: update tss address on connectors and custody contracts

### PR DESCRIPTION
**Pull request by kingpinXD**
# Description

- Update Erc20Custody and Connector contract addresses after migration is is complete
- Improve the TSS migration test to run the entire suite instead of just a few tests 

Closes: https://github.com/zeta-chain/node/issues/2443
https://github.com/zeta-chain/node/issues/2444

# How Has This Been Tested?

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions

## Summary by CodeRabbit

- **New Features**
  - Introduced structured testing workflows for "migrate" and "upgrade" scenarios in end-to-end tests.
  - Enhanced functionality for updating TSS addresses in Ethereum smart contracts.

- **Bug Fixes**
  - Increased gas limit for transactions in the `TestMessagePassingRevertFailExternalChains` function, improving transaction execution success.

- **Documentation**
  - Updated changelog to reflect recent changes in connector and ERC20 custody addresses for TSS migration tests.

- **Style**
  - Minor formatting improvements across various test functions for better readability.

- **Chores**
  - Increased default context timeout duration to enhance system robustness.

### CODE REVIEW

File: e2e/utils/zetacore.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ const (
 	AdminPolicyName       = "admin"
 	OperationalPolicyName = "operational"
 
-	DefaultCctxTimeout = 4 * time.Minute
+	DefaultCctxTimeout = 6 * time.Minute
```
Comment (kingpinXD) [inline e2e/utils/zetacore.go:23]: I am not sure why its needed now , but incrementing this solved the issues I have been facing when running tests on my local network

File: contrib/localnet/orchestrator/start-zetae2e.sh [hunk 2]
```diff
@@ -90,9 +90,47 @@ geth --exec "eth.sendTransaction({from: eth.coinbase, to: '${address}', value: w
 address=$(yq -r '.additional_accounts.user_migration.evm_address' config.yml)
 echo "funding migration tester address ${address} with 10000 Ether"
 geth --exec "eth.sendTransaction({from: eth.coinbase, to: '${address}', value: web3.toWei(10000,'ether')})" attach http://eth:8545
-
 ### Run zetae2e command depending on the option passed
 
+# Mode migrate is used to run the e2e tests before and after the TSS migration
+# It runs the e2e tests with the migrate flag which triggers a TSS migration at the end of the tests. Once the migrationis done the first e2e test is complete
+# The second e2e test is run after the migration to ensure the network is still working as expected with the new tss address
+if [ "$LOCALNET_MODE" == "migrate" ]; then
```
Comment (swift1337) [inline contrib/localnet/orchestrator/start-zetae2e.sh:98]: what does migrate mean in this context? I.e. the name is not descriptive as we have `LOCALNET_MODE=upgrade` which is also a "migration"

File: e2e/utils/zetacore.go (hunk 1 above)
Comment (swift1337) [reply e2e/utils/zetacore.go:23]: Would be good to track this as issue

File: contrib/localnet/orchestrator/start-zetae2e.sh (hunk 2 above)
Comment (kingpinXD) [reply contrib/localnet/orchestrator/start-zetae2e.sh]: 0e6fa1c995731728ae47b26b1f03a76ae54aa22b

File: e2e/utils/zetacore.go (hunk 1 above)
Comment (kingpinXD) [reply e2e/utils/zetacore.go:23]: https://github.com/zeta-chain/node/issues/2690

**Review by swift1337 — APPROVED** [review]

File: e2e/utils/zetacore.go (hunk 1 above)
Comment (fbac) [reply e2e/utils/zetacore.go:26]: Could be related to the fact that we introduced a couple e2e tests which in turn invoke `MustWaitForTxReceipt`.

**Review by fbac — APPROVED** [review]

File: e2e/runner/update_contracts.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
+package runner
```
Comment (lumtis) [inline e2e/runner/update_contracts.go:1]: I think we can name the file `admin_evm`

Admin actions related to the evm chain

File: cmd/zetae2e/local/local.go [hunk 4]
```diff
@@ -406,7 +406,7 @@ func waitKeygenHeight(
 	}
 }
 
-func runTSSMigrationTest(deployerRunner *runner.E2ERunner, logger *runner.Logger, verbose bool, conf config.Config) {
+func runTSSMigration(deployerRunner *runner.E2ERunner, logger *runner.Logger, verbose bool, conf config.Config) {
```
Comment (lumtis) [inline cmd/zetae2e/local/local.go:409]: We can move this function to a `tss_migration` file

File: changelog.md [hunk 5]
```diff
@@ -21,6 +21,10 @@
 
 ### Tests
 
+* [2661](https://github.com/zeta-chain/node/pull/2661) - update connector and erc20Custody addresses in tss migration e2e tests
+
+### Tests
```
Comment (lumtis) [inline changelog.md:26]: Test duplicated

File: x/crosschain/keeper/msg_server_migrate_tss_funds.go [hunk 6]
```diff
@@ -106,6 +106,12 @@ func (k Keeper) initiateMigrateTSSFundsCCTX(
 		return err
 	}
 
+	// Set the CCTX and the nonce for the outbound migration
+	err = k.SetObserverOutboundInfo(ctx, chainID, &cctx)
+	if err != nil {
+		return errorsmod.Wrap(types.ErrCannotMigrateTssFunds, err.Error())
```
Comment (lumtis) [inline x/crosschain/keeper/msg_server_migrate_tss_funds.go:112]: Can't we add unit test that covers this line?

**Review by gartnera — APPROVED** [review]

**Pull request merged**